client/components/render-search-results.jsx

```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+
+export default class RenderSearchResult extends React.Component {
+  render() {
+    const { results, addToLibrary } = this.props;
+
+    let author = '';
+    if (results.volumeInfo.authors !== undefined) {
+      author = results.volumeInfo.authors.join(', ');
+    }
+    let publishedYear = '';
+    if (results.volumeInfo.publishedDate !== undefined) {
+      publishedYear = results.volumeInfo.publishedDate.slice(0, 4);
+    }
+
+    let src = 'https://fivebooks.com/app/uploads/2010/09/no_book_cover.jpg';
+    if (results.volumeInfo.imageLinks !== undefined) {
+      src = results.volumeInfo.imageLinks.thumbnail;
+    }
```

Comment:
You can take advantage of nullish coalescing and optional chaining here.

```js
const author = results.volumeInfo.authors?.join(', ') ?? '';
const publishedYear = result.volumeInfo.publishedDate?.slice(0, 4) ?? '';
const src = results.volumeInfo.imageLinks?.thumbnail ?? 'https://.......';
```


https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_operator